### PR TITLE
Add mutli-pipeline to Gitea

### PR DESCRIPTION
**Pull request by ygbillet**
from https://woodpecker.laszlo.cloud/multi-pipeline/

Signed-off-by: Yves-Gaël BILLET <ygbillet@ifotec.com>

### CODE REVIEW

**Comment by ygbillet** [comment]
A proposition for adding multi-pipeline to Gitea.

File: remote/gitea/gitea.go [hunk 1]
```diff
@@ -295,7 +297,44 @@ func (c *client) File(u *model.User, r *model.Repo, b *model.Build, f string) ([
...
+			if err != nil {
+				errors += fmt.Sprintf("cannot get %s: %s", e.Path, err)
+			}
+
+			config := remote.FileMeta{
+				Name: e.Path,
+				Data: data,
+			}
+
+			configs = append(configs, &config)
```
Comment (6543) [inline remote/gitea/gitea.go:329]:
```suggestion
			configs = append(configs, &remote.FileMeta{
				Name: e.Path,
				Data: data,
			})
```
just a code format suggestion ...

**Comment by laszlocph** [comment]
This addresses #131 for Gitea

Anyone wants to test this?
CI passing, implementation looks sane.

**Comment by speatzle** [comment]
This Works for me, though it took me a while to figure out i need to set the Pipeline Path to .drone (the folder) and that if i have the pipelines .drone/.drone.yaml and .drone/.test.yaml the .drone.yml pipeline will run but the ui only shows the .test.yml one for me, is this a bug you can confirm?

**Comment by laszlocph** [comment]
> i need to set the Pipeline Path to .drone (the folder) and that if i have the pipelines .drone/.drone.yaml and .drone/.test.yaml

correct

> the .drone.yml pipeline will run but the ui only shows the .test.yml one for me, is this a bug you can confirm?

You mean you have the multipipes in `.drone/` and also a `.drone.yml` in root?
Hm, not sure how that should work. Needs to be checked and documented.

**Comment by speatzle** [comment]
> You mean you have the multipipes in `.drone/` and also a `.drone.yml` in root?

No i have the pipelines `.drone/.drone.yaml` and `.drone/.test.yaml` 
The ui wouldn't show me the .drone/.drone.yaml one but it would still run.

**Comment by speatzle** [comment]
@laszlocph can you test whether a .drone/.drone.yml pipeline is also invisible in the UI when using github multi-pipeline?

**Comment by ygbillet** [comment]
We need to reflect change in Dir() at [gitea_oauth.go](https://github.com/ygbillet/woodpecker/blob/gitea-multi-pipeline/remote/gitea/gitea_oauth.go#L282).

But we will have a code duplication. 
@6543  : do you have any insight ?

**Comment by 6543** [comment]
At the moment we have lot of different code and differnt code styles in remote.

In the end I'd like to refactor it to have one Interface and a base implementation, each remote do extend&overwrite.

For example: giteas webhook response have its own defined struct in woodpecker and we use the SDK, sobwe should use the Structs of the SDK since its up to date and tested ...

**Comment by ygbillet** [comment]
OK. 

Do we need more work for this PR (multipipeline to GITEA) ? 
We use it at work and it will be good not to have to maintain our own version.

**Comment by speatzle** [comment]
> We use it at work and it will be good not to have to maintain our own version.

We have the same situation, would be nice to have it in the latest image. i can make a new issue for the invisible pipeline problem, i belive that is a general multipipeline issue anyway and not specific to gitea.

**Comment by 6543** [comment]
I can merge it once I had time to give it a final "manual live" test

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
code is fine, and test worked too - didn't had a look int this ui issue but created an issue -> #239

@ygbillet would be nice if you could provide more info or even better dig bit into it :)

**Comment by ygbillet** [comment]
Thanks !

Regarding #239, i will give it a try (i'm not the one who found this bug)

**Comment by UnlimitedCookies** [comment]
I guess the documentation now needs to be amended accordingly. Currently, it still states the following:

> NOTE: Feature is only available for Github repositories. Follow this issue to support further development.
https://woodpecker.laszlo.cloud/multi-pipeline/

**Comment by 6543** [comment]
@UnlimitedCookies ->  #254

**Pull request merged**